[AV-43732] Search Developer Guide for Capella

sarahlwelton opened the pull request:
The long awaited Search developer guide. 

This is the partner PR for #42. 

You can review the documentation here: 

https://docs-staging.couchbase.com/cloud/search/search.html

sarahlwelton committed:
[AV-43732] Initial commit to new repo of search devex

sarahlwelton committed:
[AV-43732] Changing component in antora.yml

sarahlwelton committed:
[AV-43732] Changing nav to point to the correct version of files

sarahlwelton committed:
[AV-43732] Changed level depth in nav.adoc

sarahlwelton committed:
[AV-43732] Adding version coordinate to ToC

sarahlwelton committed:
[AV-43732] Moved content from create-child-field-steps.adoc into create-child-field.adoc

Added content to make it clear that the file is for Capella

sarahlwelton committed:
[AV-43732] Commit to move over all work from docs-server repo

sarahlwelton committed:
[AV-43732] Removed all instances of Couchbase Server.

Removed all content about Quick Index, as this doesn't exist in Capella.

Added introductory sentences for procedures.

Removed tabs attributes.

sarahlwelton committed:
[AV-43732] Removing version and module segments from nav.adoc

sarahlwelton committed:
[AV-43732] Readd version and component to links

sarahlwelton committed:
[AV-43732] Added code examples for autocomplete.

Added information on autocomplete.

Brought most-up-to-date versions of topics over from search-server to search-capella and edited content to fit with Capella.

sarahlwelton committed:
[AV-43732] Shortened titles for index customization topics.

Removed unnecessary example files.

Fixed typo in create-custom-date-time-parser.adoc.

Fixed typos in create-custom-token-filter.adoc

Added descriptions for some missing token filters to default-token-filters-reference.adoc

Added descriptions for all wordlists to default-wordlists-reference.adoc

Added and edited steps in search-query-auto-complete-code.adoc

Edited steps in search-query-auto-complete-ui.adoc

Added content to search-query-auto-complete.adoc

Added JSON to title of search-request-params.adoc

Added link to JSON properties for search requests to simple-search-rest-api.adoc

Added step to simple-search-ui.adoc

Changed depth of auto-complete topics in nav.adoc.

sarahlwelton committed:
[AV-43732] Final polish and review.

Lots of clarity and consistency edits.

Moved json paylods for search index into separate files for a better user experience, rather than having two indexes in one file.

Made default character filters and default tokenizers topics instead of leaving them in customize-index.adoc

Put links for SDKs into a partial to make management easier.

Rearranged and fixed some links in the nav.

sarahlwelton committed:
[AV-43732] Updated nav.adoc and antora.yml to work with new version scheme for couchbase-cloud

sarahlwelton committed:
[AV-43732] Escaping missed {}

sarahlwelton committed:
[AV-43732] Quick style correction on field names

bfavini commented on modules/search/pages/create-child-field.adoc:
```diff
@@ -0,0 +1,120 @@
+= Create a Child Field
+:page-topic-type: guide
+:description: Create a child field with the Couchbase Capella UI to add or remove a specific field's content from a Search index.
```
Do we need tosay "with the Couchbase Capella UI?"

sarahlwelton replied:
Yes, so they know that they're in the Capella Developer Guide and not Server.

sarahlwelton committed:
[AV-43732] Making equivalent changes from search-server PR review

osfameron committed:
Fix lines to tags

This is, admittedly a PITA, especially when VSCode "helpfully" messes
up the formatting of the JSON that you're working around, but apart
from being slightly awkward, it seems to work like this.

It feels like there are layered improvements to make this nicer.

1. use line numbers because tags are annoying

2. use tags because even so, they'll be more maintainable longer term

3. Looking at the requirements, it seems like we want to:

 * select a path from the JSON
 * elide after a certain number of lines, because the output

 So something like this syntax for an include processor might be nice?

 `include::some-json[path=.params.mapping, elide=12]`

4. It looks like this is a very similar requirement to the Mobile
  schema I'm working on, see e.g.
  https://docs.couchbase.com/sync-gateway/current/configuration-schema-database.html#_database

  (Takes a second to appear after loading, as it's generated
  dynamically)
  Notice how the key names are linked to the definitions below.

  Not quite identical use-case, but I wonder if this is something that
  we could look at generalizing?

osfameron committed:
Reindent JSON

sarahlwelton committed:
[AV-43732] Convert line includes to tags

sarahlwelton committed:
[AV-43732] Adding some changes that got missed from PR#42

osfameron reviewed (approved):
Giving unblocking approval based on review of #42

sarahlwelton committed:
[AV-43732] Adding in changes that got missed from PR#42
